relay-server/src/services/processor/dynamic_sampling.rs

```diff
@@ -71,6 +72,29 @@ pub fn validate_and_set_dsc(
     sampling_project_info
 }
 
+fn apply_legacy_sample_rate(
+    managed_envelope: &mut TypedEnvelope<TransactionGroup>,
+    event: &mut Annotated<Event>,
+) {
+    let Some(dsc) = managed_envelope.envelope().dsc() else {
+        return;
+    };
+
+    // Don't override sample rates already given in the DSC.
+    if dsc.sample_rate.is_some() {
```

Comment:
Should we also override if we got the default (0.0) value?
```suggestion
    if dsc.sample_rate.is_some_and(|rate| rate > 0.0) {
```
Seems safer only doing this when `sample_rate` is `None`, but I've seen us use this pair of checks to determine if sample rate is set elsewhere.

Reply:
I think just checking for `Some` is the safer bet here, if someone explicitly sends `0.0` it is already not an old SDK, which would be the case you want to check here for.

Other places checking for this, seems honestly a bit weird and almost like a mistake. With a `0` sample rate we still might need the data to extract it into metrics, so not doing things based on the sample rate seems strange.

Reply:
Agreed :+1: Thanks!